src/Config.cpp

```diff
@@ -437,16 +437,52 @@ bool getStatusConfig()
 // ************************************************************
 void generateSerial(bool force)
 {
-    MFeeprom.read_block(MEM_OFFSET_SERIAL, serial, MEM_LEN_SERIAL);
-    if (!force && serial[0] == 'S' && serial[1] == 'N')
+    // A serial number according old style is already generated and saved to the eeprom
+    // So keep it to avoid a connector message with orphaned board
+    if (!force && MFeeprom.read_byte(MEM_OFFSET_SERIAL) == 'S' && MFeeprom.read_byte(MEM_OFFSET_SERIAL + 1) == 'N') {
+        MFeeprom.read_block(MEM_OFFSET_SERIAL, serial, MEM_LEN_SERIAL);
+        return;
+    }
+
+    // A serial number according the uniqueID is already available, so read it
+    if (!force && MFeeprom.read_byte(MEM_OFFSET_SERIAL) == 'I' && MFeeprom.read_byte(MEM_OFFSET_SERIAL + 1) == 'D') {
+        sprintf(serial, "SN-");
+        for (size_t i = 0; i < UniqueIDsize; i++) {
+            if (UniqueID[i] < 0x10) {
+                sprintf(&serial[3 + i * 2], "0%X", UniqueID[i]);
+            } else {
+                sprintf(&serial[3 + i * 2], "%X", UniqueID[i]);
+            }
+        }
+        return;
+    }
+
+    // Coming here it's the first start up of the board and no serial number or UniqueID is available
+    // Read the uniqueID and use it as serial numnber
+    if (!force) {
+        sprintf(serial, "SN-");
+        for (size_t i = 0; i < UniqueIDsize; i++) {
+            if (UniqueID[i] < 0x10) {
+                sprintf(&serial[3 + i * 2], "0%X", UniqueID[i]);
+            } else {
+                sprintf(&serial[3 + i * 2], "%X", UniqueID[i]);
+            }
+        }
+        // and mark this in the eeprom
+        MFeeprom.write_block(MEM_OFFSET_SERIAL, "ID", 2);
+        // Set first byte of config to 0x00 to ensure with empty config on 1st start up
+        MFeeprom.write_byte(MEM_OFFSET_CONFIG, 0x00);
```

Comment:
The only difference between these two blocs is whether the word "ID" gets written to eeprom. There's no real downside to writing "ID" every time this is called.

Combined with my prior comment about `force` being tested first I believe that would simplify this function down to three cases:

1. force is true
2. SN is in eeprom so there's an old-style ID on the board
3. all other cases